api/src/main/java/org/apache/iceberg/DeletedDataFileScanTask.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+
+/**
+ * A scan task for deleted data records generated by removing a data file from the table.
+ * <p>
+ * Note that all historical delete files added before must be applied while reading the data file to
+ * output only those data records that were live when the data file was removed.
+ */
+public interface DeletedDataFileScanTask extends ChangelogScanTask, ContentScanTask<DataFile> {
+  /**
+   * A list of {@link DeleteFile delete files} to apply when reading the task's data file.
+   *
+   * @return a list of delete files to apply
+   */
+  List<DeleteFile> deletes();
```

Comment:
I feel like this is related to our "existingDeletes" in the DeletedRowsScan task than it is to `deletes` in `AddedRowScanTask`. Maybe worth changing the name?

Just trying to differentiate between files marking rows which have been deleted so we aren't rendering them and files marking rows which were deleted in the past so we aren't rendering them.

Reply:
Changed.